lib/iris/tests/unit/representation/test_representation.py

```diff
@@ -182,6 +182,56 @@ def test_cell_methods(self):
         expected_contents = ["mean: x, y", "mean: x"]
         self.assertEqual(cell_method_section.contents, expected_contents)
 
+    def test_scalar_cube(self):
+        cube = self.cube
+        while cube.ndim > 0:
+            cube = cube[0]
+        rep = iris._representation.CubeSummary(cube)
+        self.assertEqual(rep.header.nameunit, "air_temperature / (K)")
+        self.assertTrue(rep.header.dimension_header.scalar)
+        self.assertEqual(rep.header.dimension_header.dim_names, [])
+        self.assertEqual(rep.header.dimension_header.shape, [])
+        self.assertEqual(rep.header.dimension_header.contents, ["scalar cube"])
+        self.assertEqual(len(rep.vector_sections), 5)
+        self.assertTrue(
+            all(sect.is_empty() for sect in rep.vector_sections.values())
+        )
+        self.assertEqual(len(rep.scalar_sections), 4)
+        self.assertEqual(
+            len(rep.scalar_sections["Scalar coordinates:"].contents), 1
+        )
+        self.assertTrue(
+            rep.scalar_sections["Scalar cell measures:"].is_empty()
+        )
+        self.assertTrue(rep.scalar_sections["Attributes:"].is_empty())
+        self.assertTrue(rep.scalar_sections["Cell methods:"].is_empty())
+
+    def test_coord_attributes(self):
+        cube = self.cube
+        co1 = cube.coord("latitude")
+        co1.attributes.update(dict(a=1, b=2))
+        co2 = co1.copy()
+        co2.attributes.update(dict(a=7, z=77))
+        cube.add_aux_coord(co2, cube.coord_dims(co1))
+        rep = iris._representation.CubeSummary(cube)
+        co1_summ = rep.vector_sections["Dimension coordinates:"].contents[0]
+        co2_summ = rep.vector_sections["Auxiliary coordinates:"].contents[0]
+        self.assertEqual(co1_summ.extra, "a=1")
+        self.assertEqual(co2_summ.extra, "a=7, z=77")
+
+    def test_array_attributes(self):
+        cube = self.cube
+        co1 = cube.coord("latitude")
+        co1.attributes.update(dict(a=1, array=np.array([1.2, 3])))
+        co2 = co1.copy()
+        co2.attributes.update(dict(b=2, array=np.array([3.2, 1])))
+        cube.add_aux_coord(co2, cube.coord_dims(co1))
+        rep = iris._representation.CubeSummary(cube)
+        co1_summ = rep.vector_sections["Dimension coordinates:"].contents[0]
+        co2_summ = rep.vector_sections["Auxiliary coordinates:"].contents[0]
+        self.assertEqual(co1_summ.extra, "array=array([1.2, 3. ])")
+        self.assertEqual(co2_summ.extra, "array=array([3.2, 1. ]), b=2")
```

Comment:
For the sake of testing numpy array equality,  it might also be worth testing the case where two arrays are equal and the case where arrays have different shapes.